### PR TITLE
Branch 192

### PR DESCRIPTION
**Pull request by NikhitaPaturu**
email notification

### CODE REVIEW

File: app/controller/partycontroller.ctrl.php [hunk 1]
```diff
@@ -201,10 +201,29 @@ public function create()
                         }
 
                         if(hasRole($this->user, 'Host')){
+                            // send email to Admin
+                              $message = "<p>Hi,</p>" .
+                                       "<p>This is an automatic email to let you know that ". $name . " has created a party on the <strong>" . APPNAME . "</strong>.</p>" .
+                                        "<p><strong>Group Name:</strong> ". $group ." <p>" .
```
Comment (ngm) [inline app/controller/partycontroller.ctrl.php]: $group is the group id (we should really rename this variable to $group_id...)
For group name, you will need to do something like:

    $group_obj = $Groups->findOne($group);
    $group_name = $group_obj->name;

File: app/controller/partycontroller.ctrl.php [hunk 2]
```diff
@@ -201,10 +201,29 @@ public function create()
                         }
 
                         if(hasRole($this->user, 'Host')){
+                            // send email to Admin
+                              $message = "<p>Hi,</p>" .
+                                       "<p>This is an automatic email to let you know that ". $name . " has created a party on the <strong>" . APPNAME . "</strong>.</p>" .
```
Comment (ngm) [inline app/controller/partycontroller.ctrl.php]: Where is this `$name` variable defined?
This should come from $this->user->name I think.

File: app/controller/partycontroller.ctrl.php [hunk 3]
```diff
@@ -201,10 +201,29 @@ public function create()
...
+                                       "<p><strong>Party Location:</strong> " . $location ." </p>" .
+                                       "<p><strong>Party Date:</strong> ". $event_date ." </p>" .
+                                       "<p><strong>Party Start Time:</strong> ". $start ." </p>" .
+                                       "<p><strong>Party End Time:</strong> ". $end ." </p>" .
+                                       "<p> Thank you!!! <p>";
+
+                              $subject = APPNAME . ": Party created by the host : " . user . " ";
+                              $headers = "From: " . APPEMAIL . "\r\n";
+                              $headers .= "Content-Type: text/html; charset=ISO-8859-1\r\n";
+                              $email= "Papercut@user.com";
```
Comment (ngm) [inline app/controller/partycontroller.ctrl.php]: We should define a constant in config.php for this, named NOTIFICATION_EMAIL.

File: app/controller/partycontroller.ctrl.php (hunk 2 above)
Comment (NikhitaPaturu) [reply app/controller/partycontroller.ctrl.php]: Its defined in usercontroller.

Comment (ngm) [reply app/controller/partycontroller.ctrl.php]: It won't be accessible here - you will need to use the `$this->user` object.

File: app/controller/partycontroller.ctrl.php [hunk 4]
```diff
@@ -201,10 +201,35 @@ public function create()
                         }
 
                         if(hasRole($this->user, 'Host')){
+
+                            $group_obj = $Groups->findOne($group);
+                            $group_name = $group_obj->name;
+
+                           $hostname = $this->$user->name;
```
Comment (ngm) [inline app/controller/partycontroller.ctrl.php]: This should be `$this->user->name;` not `$this->$user->name`.

File: app/controller/partycontroller.ctrl.php [hunk 5]
```diff
@@ -218,7 +218,7 @@ public function create()
                                        "<p><strong>Party End Time:</strong> ". $end ." </p>" .
                                        "<p> Thank you!!! <p>";
 
-                              $subject = APPNAME . ": Party created by the host : " . $hostnamep . " ";
+                              $subject = APPNAME . ": Party created by the host : " . $hostname . " ";
                               $headers = "From: " . APPEMAIL . "\r\n";
                               $headers .= "Content-Type: text/html; charset=ISO-8859-1\r\n";
                               $email= NOTIFICATION_EMAIL;
```
Comment (ngm) [inline app/controller/partycontroller.ctrl.php]: Could you add this new constant in to config.example.php, with a made up email address.

**Pull request merged**